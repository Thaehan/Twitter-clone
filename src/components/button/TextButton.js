import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import {
  MAIN_COLOR,
  BACKGROUND_COLOR
} from "../../styles/Style"
// Nhận vào title, onPress, style
export default function TextButton(props) {
  return (
    <View style={props.style}>
      <TouchableOpacity
        style={styles.container}
        onPress={props.onPress}
      >
        <Text style={styles.buttonText}>{props.title}</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  buttonText: {
    color: MAIN_COLOR,
    fontWeight: '500',
    padding: 0,
  },
  container: {
    alignContent: 'center',
    alignSelf: 'center',
    borderRadius: 50,
  },
});
