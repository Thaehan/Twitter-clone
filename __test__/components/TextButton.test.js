import React from 'react';
// import renderer from 'react-test-renderer';\
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import store from '../../src/redux/store';
import TextButton from '../../src/components/button/TextButton';

jest.useFakeTimers();
jest.mock(
  'react-native/Libraries/Animated/NativeAnimatedHelper'
);

describe('<TextButton />', () => {
  it('Render TextButton', () => {
    const testComponent = render(<TextButton />);
    expect(testComponent.toJSON()).toMatchSnapshot();
    // testComponent.getByTestId('AvatarImage').tobe();
  });
});
