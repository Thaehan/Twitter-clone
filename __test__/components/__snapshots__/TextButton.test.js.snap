// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton /> Render TextButton 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignContent": "center",
        "alignSelf": "center",
        "borderRadius": 50,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontWeight": "500",
            "padding": 0,
          },
          Object {
            "color": undefined,
          },
        ]
      }
    />
  </View>
</View>
`;
